app/src/main/java/io/apicurio/registry/storage/RegistryStorage.java

```diff
@@ -169,10 +169,11 @@ public ArtifactMetaDataDto createArtifactWithMetadata(String groupId, String art
      * Gets the most recent version of the value of the artifact with the given group and ID.
      * @param groupId (optional)
      * @param artifactId
+     * @param skipDisabled Skip artifact versions with state=DISABLED metadata
      * @throws ArtifactNotFoundException
      * @throws RegistryStorageException
      */
-    public StoredArtifactDto getArtifact(String groupId, String artifactId) throws ArtifactNotFoundException, RegistryStorageException;
+    public StoredArtifactDto getArtifact(String groupId, String artifactId, boolean skipDisabled) throws ArtifactNotFoundException, RegistryStorageException;
```

Comment:
Perhaps instead of a "skipDisabled" boolean flag, some other structured indication of the states to include/exclude.  That would make this more flexible for the future **and** it would improve the readability of any code invoking this method (vs. this approach that looks like `storage.getArtifact(group, artifactId, true)` where a dev might need to wonder what `true` meant.

Reply:
Agreed, using the boolean was a bit confusing and is the exact reason for me to delay the review :). Having a collection of states to skip that can eventually be loaded from an env var makes a lot of sense to me.